kernel/Kconfig

```diff
@@ -75,13 +75,17 @@ config NUM_PREEMPT_PRIORITIES
 	  priority, and be the only thread at that priority.
 
 config MAIN_THREAD_PRIORITY
-	int "Priority of initialization/main thread"
+	int
+	prompt "Priority of initialization/main thread" if MAIN_THREAD_PRIORITY_WITH_PROMPT
 	default -2 if !PREEMPT_ENABLED
 	default 0
 	help
 	  Priority at which the initialization thread runs, including the start
 	  of the main() function. main() can then change its priority if desired.
 
+config MAIN_THREAD_PRIORITY_WITH_PROMPT
```

Comment:
Another option might be `CUSTOM_MAIN_THREAD_PRIORITY` (as opposed to one that gets set automatically).

Best to think of Kconfig as a simple options menu. You check `Custom thread priority`, then file in the value.

Reply:
Perhaps, but I'd like to be consistent with https://github.com/zephyrproject-rtos/zephyr/pull/20702 , and I'd prefer not to change https://github.com/zephyrproject-rtos/zephyr/pull/20702 because that would break compatibility.

Reply:
IMHO `CUSTOM_MAIN_THREAD_PRIORITY` is a better name. The `_WITH_PROMPT` assumes user is a Kconfig guru and understands ramifications of symbol having or not having a prompt. It's also not that clear that we are talking about Kconfig prompt. When using menuconfig user will see the text description, which is fine, but having this symbol set in prj.conf surrounded by a bunch other Kconfig options will make it cryptic.

On the other hand being consistent with #20702 is also a valid argument.